airflow/utils/file.py

```diff
@@ -335,7 +335,23 @@ def find_dag_file_paths(directory: str | pathlib.Path, safe_mode: bool) -> list[
 COMMENT_PATTERN = re.compile(r"\s*#.*")
 
 
-def might_contain_dag(file_path: str, safe_mode: bool, zip_file: zipfile.ZipFile | None = None):
+def might_contain_dag(file_path: str, safe_mode: bool, zip_file: zipfile.ZipFile | None = None) -> bool:
```

Comment:
I think safe_mode is not needed. IMHO this check if should be done before running callable.

As it is now - every callable should start with:

```
    if not safe_mode:
        return True
```

And we should do it even before importing the callable.

Reply:
that is a good idea. in this way, we don't need to expose `safe_mode` to the callable as part of the interface